OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -179,7 +186,27 @@ public void Tick(Actor self)
 		public virtual IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr)
 		{
 			return world.ScreenMap.FrozenActorsInBox(owner, wr.Viewport.TopLeft, wr.Viewport.BottomRight)
-				.Where(f => f.Visible)
+				.Where(f =>
```

Comment:
This sounds much more complicated than it needs to be.  A simpler solution would be to change FrozenActor to generate the sprites from an ActorPreview if the actor has been destroyed.